dbms/src/Storages/DeltaMerge/tests/gtest_sst_files_stream.cpp

```diff
@@ -464,6 +492,59 @@ try
 }
 CATCH
 
+TEST_F(SSTFilesToDTFilesOutputStreamTest, UpperLayerCancel)
+try
+{
+    auto table_lock = storage->lockStructureForShare("foo_query_id");
+    auto [schema_snapshot, unused] = storage->getSchemaSnapshotAndBlockForDecoding(table_lock, false);
+
+    auto mock_stream = makeMockChild(prepareBlocks(50, 100, /*block_size=*/1));
+    std::shared_ptr<std::atomic_bool> abort_flag = std::make_shared<std::atomic_bool>(false);
+    auto stream = std::make_shared<DM::SSTFilesToDTFilesOutputStream<DM::MockSSTFilesToDTFilesOutputStreamChildPtr>>(
+        /* log_prefix */ "",
+        mock_stream,
+        storage,
+        schema_snapshot,
+        FileConvertJobType::ApplySnapshot,
+        /* split_after_rows */ 10,
+        /* split_after_size */ 0,
+        0,
+        abort_flag,
+        *db_context);
+
+    auto sp = SyncPointCtl::enableInScope("before_SSTFilesToDTFilesOutputStream::handle_one");
+    stream->writePrefix();
+    auto t = std::thread([&]() {
+        stream->write();
+    });
+    sp.waitAndPause();
+    abort_flag->store(true, std::memory_order_seq_cst);
+    sp.disable();
+    t.join();
+    stream->writeSuffix();
+    auto files = stream->outputFiles();
```

Comment:
Add check for `stream->isAborted() == true`

Reply:
fixed